### PR TITLE
{Packaging} Change GitHub artifact URL to `/archive/refs/tags`

### PR DESCRIPTION
**Pull request by chenrui333**
**Description**
Update github artifact urls.

relates to 
- https://github.com/Homebrew/brew/pull/16126
- https://github.com/Homebrew/homebrew-core/pull/152239

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by chenrui333** [comment]
Thanks @yonzhan!

File: scripts/release/homebrew/upload.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ CLI_VERSION=`cat $SYSTEM_ARTIFACTSDIRECTORY/metadata/version`
 TAR_NAME=azure-cli-$CLI_VERSION.tar.gz
 
 mkdir -p $BUILD_STAGINGDIRECTORY/github/ >/dev/null 2>&1
-curl -sL https://github.com/Azure/azure-cli/archive/$TAR_NAME -o $BUILD_STAGINGDIRECTORY/github/$TAR_NAME
+curl -sL https://github.com/Azure/azure-cli/archive/refs/tags/$TAR_NAME -o $BUILD_STAGINGDIRECTORY/github/$TAR_NAME
```
Comment (bebound) [inline scripts/release/homebrew/upload.sh:14]: I think this file is deprecated. Could you please revert this change? @jiasli for confirmation.

---------------

In https://github.com/Azure/azure-cli/pull/8178, we uses GitHub url as `HOMEBREW_UPSTREAM_URL`, so `scripts/release/homebrew/upload.sh` is useless.

We also move the formula generation logic to https://github.com/Azure/azure-cli/blob/22feb5a5dcb1e23f0c1f1a444ff99af5c5502b19/azure-pipelines.yml#L597-L610
, so `scripts/release/homebrew/pipeline.sh` can also be removed.

**Comment by bebound** [comment]
Thank you, I've upload the related logic in our release pipline.
``` bash
cat ${SYSTEM_DEFAULTWORKINGDIRECTORY}/_Azure.azure-cli/homebrew/azure-cli.rb > Formula/a/azure-cli.rb
SHA=`curl -L https://github.com/Azure/azure-cli/archive/refs/tags/azure-cli-${CLI_VERSION}.tar.gz | sha256sum | awk '{ print $1 }'`
# replace the first occurence of url
sed -i '0,/^  url.*$/s||  url "https://github.com/Azure/azure-cli/archive/refs/tags/azure-cli-'"${CLI_VERSION}"'.tar.gz"|' Formula/a/azure-cli.rb
```

**Pull request closed without merging**